app/util/DataLogging.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package util
+
+import play.api.libs.json.Json
+import play.api.mvc.RequestHeader
+import ssttparrangement.SubmissionError
+import uk.gov.hmrc.selfservicetimetopay.models.{PaymentPlanRequest, TTPArrangement}
+
+object DataLogging {
+  trait ArrangementLogging {
+    self: BaseLogger =>
+
+    def debug(message: => String, arrangement: TTPArrangement)(implicit request: RequestHeader): Unit = logMessage(message, arrangement, Debug)
+
+    def info(message: => String, arrangement: TTPArrangement)(implicit request: RequestHeader): Unit = logMessage(message, arrangement, Info)
+
+    def warn(message: => String, arrangement: TTPArrangement)(implicit request: RequestHeader): Unit = logMessage(message, arrangement, Warn)
+
+    def error(message: => String, arrangement: TTPArrangement)(implicit request: RequestHeader): Unit = logMessage(message, arrangement, Error)
+
+    private def appendedArrangement(arrangement: TTPArrangement): String = s" [arrangement: ${appendedData(Json.toJson(arrangement.obfuscate))}]"
```

Comment:
here and below, I'm not sure we really to ever log details of an arrangement - if we need the info we can look in splunk. We are obfuscating here but it feels like the kibana logs shouldn't have this level of detail in it.

Reply:
Ok. Currently we would log the obfuscated arrangement if the arrangement set up didn't work (i.e. the DD instruction has been successful but arrangement hasn't gone through and Debt Management would then have to manually set up the arrangement). The arrangement included the payment plan reference and the direct debit reference.

We also audit this and the audit also has the payment plan reference and direct debit reference.

This allows us to cross reference the audit event and the log.

If we remove the arrangement from the log, without including the payment plan reference and/or direct debit reference, we won't be able to cross check the logs against the audits.

Is that fine? Or shall I include the references in the log message?

Reply:
I have decided to remove the arrangement from the logging but to include the payment plan reference and direct debit reference for ease of cross referencing with splunk for the purposes to investigation if issues arise.